testing/e2e/infra/provision.go

```diff
@@ -124,6 +137,11 @@ func (i *infra) Provision(ctx context.Context, tenantId, subscriptionId string)
 	// connect permissions
 	var permEg errgroup.Group
 
+	roleAssignmentPrincipalId := ret.Cluster.GetPrincipalId()
+	if i.ServicePrincipalOptions != nil {
```

Comment:
why do we need this logic here instead of building it into the aks client? GetPrincipalId() can return the necessary principalId itself. We should move this logic there